### ISSUE TITLE
Implement 'hints' method to support downstream tooling

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/ophyd#413]
Most built-in devices will get methods like:

```python
class EpicsMotor:
    ...
    def hints():
        return {'axes': [self.readback.name]}
```

For example, a plotting tool can use this information (along with other information provided by the bluesky plans) to figure out what to plot by default.

**Comment by danielballan** [comment]
@ericdill We finally figured out a way to do "plot y" that seems sufficiently general. You have been vindicated about 2.5 years too late.

**Comment by tacaswell** [comment]
"while here also update read_attr and conf_attrs" --danielballan

**Comment by danielballan** [comment]
Closed by #426
